src/modules/UI/scenes/Plugins/EdgeProvider.js

```diff
@@ -339,19 +339,17 @@ export class EdgeProvider extends Bridgeable {
       return Promise.reject(e)
     }
   }
+  async signMessage (message: string) /* EdgeSignedMessage */ {
+    const guiWallet = UI_SELECTORS.getSelectedWallet(this._state)
+    const coreWallet = CORE_SELECTORS.getWallet(this._state, guiWallet.id)
 
-  // Sign a message using a public address from the current wallet
-  /* signMessage (options: EdgeSignMessageOptions): EdgeSignedMessage {
-    console.log('a1: signMessage', options)
-    // this is about bit id signatures.
-    const obj = {
-      publicKey: 'string',
-      // Hex encoded signature
-      signedMessage: 'string'
+    try {
+      const signedMessage = await coreWallet.otherMethods.signMessageBase64(message, guiWallet.receiveAddress.publicAddress)
+      return signedMessage
+    } catch (e) {
+      console.log('EP: E', e)
```

Comment:
This should be:

```js
catch (e) {
  showError(e)
  throw e // <- This way Bity knows it failed? Unless they know some other way?
}
```

Reply:
thanks, however Bity handles calling the `showError`. I figure it is better to let the individual plugins decide what to do with an error, and then if they want to use ours they can all `EdgeProvider` back and we can then show it.